### DISCUSSION TITLE
Completely Replace HTTPResponse handler

### DISCUSSION THREAD

**Discussion by qhenkart** [aws/aws-sdk-js-v3#3976] [Q&A]
I have an AWS transcription service. The FE starts the data stream to our API so we can track and limit the usage, our API passes the response to AWS. On the response, our API modifies the response data to a simpler data structure, stores it in the database and passes it back to the FE.

The problem is expecting a certain response from AWS, and will not accept a custom response. How can I replace this layer? I've been messing around with the middleware stack but the documentation around existing middleware and how to replace/remove it is really lacking

**Comment by yenfryherrerafeliz** [comment]
Hi @qhenkart, thanks for opening this discussion. When instancing an AWS client you have the option to pass a custom http handler to the client, the parameter is requestHandler. Small example below:

```
import {S3Client, GetObjectCommand} from "@aws-sdk/client-s3";
import {NodeHttpHandler} from "@aws-sdk/node-http-handler";

const client = new S3Client({
    region: 'us-east-2',
    requestHandler: new NodeHttpHandler(),
});
const response = await client.send(new GetObjectCommand({
    Bucket: 'herrergy-sample-bucket',
    Key: 'key'
}));
console.log(response);

```

[Here](https://github.com/aws/aws-sdk-js-v3/blob/6b6830595d72a7e40f6b8de39c3c8866940cbab1/packages/node-http-handler/src/node-http-handler.ts) you can find the implementation of node http handler, so you can use it as reference and guidance.

Thanks!
